src/extension/src/azure/azureServices.ts

```diff
@@ -210,10 +219,40 @@ export class AzureServices extends WizardServant {
 
     return {
       resourceGroups: await resourceGroupItems,
-      locations: locations
+      locations: locations,
+      validName: validName
     };
   }
 
+  public static async validateNameForAzureType(
+    projectName: string,
+    userSubscriptionItem: SubscriptionItem,
+    azureType: AzureResourceType
+  ): Promise<boolean> {
+    let validationResult;
+    switch (azureType) {
+      case AzureResourceType.AppService:
+        validationResult = await AzureServices.AzureAppServiceProvider.checkWebAppName(
+          projectName,
+          userSubscriptionItem
+        );
+        break;
+      case AzureResourceType.Cosmos:
+        validationResult = await AzureServices.AzureCosmosDBProvider.validateCosmosDBAccountName(
+          projectName,
+          userSubscriptionItem
+        );
+        break;
+      case AzureResourceType.Functions:
+        validationResult = await AzureServices.AzureFunctionProvider.checkFunctionAppName(
+          projectName,
+          userSubscriptionItem
+        );
+        break;
+    }
+    return validationResult ? false : true;
```

Comment:
this looks confusing.. just for context, the individual methods that checks if the name is valid return `undefined` if the name is valid, else, it would return an error message. so I was just trying to check if `validationResult` is `undefined`

Reply:
maybe it's better to have `validationResult === undefined` then?

Reply:
that works too